paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -174,21 +175,26 @@ internal class PaymentSheetViewModel @Inject internal constructor(
             }
         }
 
-    // Whether the top container, containing Google Pay and Link buttons, should be visible
-    internal val showTopContainer = MediatorLiveData<Boolean>().apply {
-        listOf(
-            linkHandler.isLinkEnabled.asLiveData(),
-            googlePayState.asLiveData(),
-            isReadyEvents
-        ).forEach {
-            addSource(it) {
-                value = (
-                    linkHandler.isLinkEnabled.value == true ||
-                        googlePayState.value == GooglePayState.Available
-                    ) && isReadyEvents.value?.peekContent() == true
-            }
-        }
-    }
+    internal val topContainerState: StateFlow<TopContainerState> = combine(
+        linkHandler.isLinkEnabled,
+        googlePayState,
+        isReadyEvents.asFlow(),
+        supportedPaymentMethodsFlow,
+    ) { isLinkAvailable, googlePayState, isReady, paymentMethodTypes ->
+        TopContainerState(
+            showLink = isLinkAvailable == true && isReady.peekContent() == true,
+            showGooglePay = googlePayState.isReadyForUse && isReady.peekContent() == true,
+            dividerTextResource = if (paymentMethodTypes.singleOrNull() == Card.code) {
+                R.string.stripe_paymentsheet_or_pay_with_card
+            } else {
+                R.string.stripe_paymentsheet_or_pay_using
+            },
+        )
+    }.stateIn(
```

Comment:
Does this have to be a StateFlow? Can this be a Flow?

Reply:
It can be. I do like the cleaner call site from using `StateFlow`, but I’m fine with making it a `Flow` to discourage one-off access.